File: trunk/omplapp/demos/SE3RigidBodyPlanning/SE3RigidBodyPlanning.cpp
#include <omplapp/config.h>
#include <omplapp/SE3RigidBodyPlanning.h>

using namespace ompl;

int main()
{
    app::SE3RigidBodyPlanning setup;
    std::string robot_fname = std::string(OMPL_RESOURCE_DIR) + "/alpha_robot.dae";
    std::string env_fname = std::string(OMPL_RESOURCE_DIR) + "/alpha_env.dae";
    setup.setRobotMesh(robot_fname.c_str());
    setup.setEnvironmentMesh(env_fname.c_str());


    base::ScopedState<base::SE3StateManifold> start(setup.getSpaceInformation());
    start->setX(-4.96);
    start->setY(70.57);
    start->setZ(40.62);
    start->rotation().setIdentity();

    base::ScopedState<base::SE3StateManifold> goal(start);
    goal->setX(200.49);
    goal->setY(70.57);
    goal->setZ(40.62);
    goal->rotation().setIdentity();

    setup.setStartAndGoalStates(start, goal);
    setup.getSpaceInformation()->setStateValidityCheckingResolution(0.01);
    setup.setup();
    
    setup.print();
    if (setup.solve(60))
    {
	setup.simplifySolution();
	setup.getSolutionPath().print(std::cout);
    }
    
    return 0;
}

File: trunk/omplapp/demos/SE3RigidBodyPlanning/SE3RigidBodyPlanning.py
#!/usr/bin/env python

######################################################################
# Rice University Software Distribution License
#
# Copyright (c) 2010, Rice University
# All Rights Reserved.
#
# For a full description see the file named LICENSE.
#
######################################################################

# Author: Mark Moll

import sys
from os.path import abspath, dirname, join

ompl_app_root = dirname(dirname(dirname(abspath(__file__))))
ompl_resources_dir = join(ompl_app_root, 'resources')

try:
    from ompl import base as ob
    from ompl import geometric as og
    from ompl import app as oa
except:
    sys.path.insert(0, join(ompl_app_root, 'ompl/py-bindings' ) )
    from ompl import base as ob
    from ompl import geometric as og
    from ompl import app as oa


setup = oa.SE3RigidBodyPlanning()
print setup.getStateManifold().getBounds().low[0]
setup.setRobotMesh(join(ompl_resources_dir, 'Twistycool_robot.dae'))
setup.setEnvironmentMesh(join(ompl_resources_dir, 'Twistycool_env.dae'))

start = ob.State(setup.getSpaceInformation())
rot = start().rotation()
(rot.w, rot.x, rot.y, rot.z) = (1,0,0,0)
start().setX(0)
start().setY(0)
start().setZ(0)

goal = ob.State(setup.getSpaceInformation())
rot = goal().rotation()
(rot.w, rot.x, rot.y, rot.z) = (1,0,0,0)
goal().setX(0)
goal().setY(0)
goal().setZ(100)

setup.setStartAndGoalStates(start, goal);
print start, goal

if setup.solve():
    setup.simplifySolution()
    path = setup.getSolutionPath()
    path.interpolate(10)
    print path, path.check()
